kloppy/infra/serializers/tracking/statsperform.py

```diff
@@ -0,0 +1,271 @@
+                team = Team(
+                    team_id=team_info.attrib["id"],
+                    name=team_info.attrib["name"],
+                    ground=ground,
+                )
+                teams.append(team)
+
+            # Get Frame Rate - Either 10FPS or 25FPS
+            frame_rate = self.__get_frame_rate(raw_data)
+            pitch_size_length = 100
+            pitch_size_width = 100
+
+            periods = self.__get_periods(raw_data)
+            # Get Player Info:
+            for line_up in objectify.fromstring(
+                metadata
+            ).liveData.iterchildren(tag="lineUp"):
+                count = 0
+                for team in teams:
+                    if team.team_id == line_up.get("contestantId"):
+                        team_index = count
+                    count += 1
+
+                for player in line_up.iterchildren(tag="player"):
+                    player = player.attrib
+                    player = Player(
+                        player_id=player["playerId"],
+                        name=player["matchName"],
+                        starting=player["position"] == "substitute",
+                        position=player["position"],
+                        team=team,
+                        jersey_no=player["shirtNumber"],
+                        attributes={},
+                    )
+                    teams[team_index].players.append(player)
+
+        # Handles the tracking frame data
+        with performance_logging("Loading raw data", logger=logger):
+            transformer = self.get_transformer(
+                length=pitch_size_length, width=pitch_size_width
+            )
+
+            def _iter():
+                n = 0
+                sample = 1.0 / self.sample_rate
+
+                for line_ in raw_data.decode("ascii").splitlines():
+                    frame_id = int(line_.split(";")[1].split(",")[0])
+                    if self.only_alive and not line_.endswith("Alive;:"):
+                        continue
+                    period_id = int(line_.split(";")[1].split(",")[1])
+                    period_ = periods[period_id - 1]
+                    if period_.contains(frame_id / frame_rate):
+                        if n % sample == 0:
+                            yield period_, line_
+                        n += 1
+
+            frames = []
+            for n, frame_data in enumerate(_iter()):
+                period = frame_data[0]
+                frame = self._frame_from_framedata(teams, period, frame_data)
+                frame = transformer.transform_frame(frame)
+                frames.append(frame)
+
+                if not period.attacking_direction_set:
+                    period.set_attacking_direction(
+                        attacking_direction=attacking_direction_from_frame(
+                            frame
+                        )
+                    )
+
+                if self.limit and n + 1 >= self.limit:
```

Comment:
Small suggestion to improve the readability: Add `start=1` to the `enumerate` call and change the implementation to `and n >= self.limit`.